docs/gallery/general/advanced_hdf5_io.py

```diff
@@ -216,6 +216,56 @@
 #   will be ignored as the h5py.Dataset will either be linked to or copied as on write.
 #
 
+####################
+# Parallel IO using MPI
+# -----------
+#
+# The HDF5 storage backend supports parallel I/O using the Message Passing Interface (MPI).
+# Using this feature requires that you install ``hdf5`` and ``h5py`` against an MPI driver, and you
+# install ``mpi4py``. The basic installation of pynwb will not work. Setup can be tricky, and
+# is outside the scope of this tutorial (for now), and the following assumes that you have
+# HDF5 installed in a MPI configuration. Here we:
+#
+# 1. Instantiate a dataset for parallel write
+#
+# 2. Write to that file in parallel using MPI
+#
+# 3. Read from the file in parallel using MPI.
+#
+# .. code-block:: python
+#
+#   from mpi4py import MPI
+#   import numpy as np
+#   from dateutil import tz
+#   from pynwb import NWBHDF5IO, NWBFile, TimeSeries
+#   from datetime import datetime
+#   from hdmf.data_utils import DataChunkIterator
+#
+#   start_time = datetime(2018, 4, 25, 2, 30, 3, tzinfo=tz.gettz('US/Pacific'))
+#   fname = 'test_parallel_pynwb.nwb'
+#   rank = MPI.COMM_WORLD.rank  # The process ID (integer 0-3 for 4-process run)
+#
+#   # instantiate dataset, but do not write any data
+#   if not rank:
+#       nwbfile = NWBFile('aa', 'aa', start_time)
+#       data = DataChunkIterator(data=None, maxshape=(4,), dtype=np.dtype('int'))
+#
+#       nwbfile.add_acquisition(TimeSeries('ts_name', description='desc', data=data,
+#                                          rate=100., unit='m'))
+#       with NWBHDF5IO(fname, 'w') as io:
+#           io.write(nwbfile)
+#
+#   # write to dataset in parallel
+#   with NWBHDF5IO(fname, 'a', comm=MPI.COMM_WORLD) as io:
+#       nwbfile = io.read()
+#       print(rank)
+#       nwbfile.acquisition['ts_name'].data[rank] = rank
+#
+#   # read from dataset in parallel
+#   with NWBHDF5IO(fname, 'r', comm=MPI.COMM_WORLD) as io:
+#       print(io.read().acquisition['ts_name'].data[:])
```

Comment:
Similar to the parallel write section, I think we should do ``nwbfile = io.read()`` first. Otherwise we end up parsing the file twice here.